### ISSUE TITLE
ExecuteGraphCommand.GetOutputToPath should not throw if the outputs are null 

### ISSUE THREAD

**Issue by sfilipi** [dotnet/machinelearning#315]
### System information

- **.NET Version (eg., dotnet --info)**: 0.3.0-preview-26604-1

### Issue

Construct an entry-points graph invoking the CV macro, and specifying the Warnings output of the CV macro, as the overall graph output. 

Notice that if the run produced no warnings, (warnings idv is null) ExecuteGraphCommand.GetOutputToPath throws. I think it should just warn and proceed to saving the rest of the output. 

### Source code / logs

Graph used:
```json
{
	"Inputs": {
		"TrainFile": "..\\Sent140_Train.tsv"
	},
	"Outputs": {
		"PredictorModel0": "PredictorModel0.zip",
		"Warnings0": "Warnings0.csv",
		"OverallMetrics0": "OverallMetrics0.csv",
		"PerInstanceMetrics0": "PerInstanceMetrics0.csv",
		"ConfusionMatrix0": "ConfusionMatrix0.csv"
	},
	"Nodes": [
		{
			"Name": "Data.CustomTextLoader",
			"Inputs": {
				"InputFile": "$TrainFile",
				"CustomSchema": " sep=tab header+ col=SentimentText:TX:3  col=Sentiment:R4:1 "
			},
			"Outputs": {
				"Data": "$TrainDataView"
			}
		},
		{
			"Name": "Models.CrossValidator",
			"Inputs": {
				"Kind": "SignatureBinaryClassifierTrainer",
				"NumFolds": 2,
				"Data": "$TrainDataView",
				"Nodes": [
					{
						"Name": "Transforms.ColumnCopier",
						"Inputs": {
							"Column": [
								{
									"Name": "Label",
									"Source": "Sentiment"
								}
							],
							"Data": "$data0"
						},
						"Outputs": {
							"OutputData": "$Copy_DataView_Output_1528237424459_579",
							"Model": "$Copy_TransformModel_Output_1528237424459_493"
						}
					},
					{
						"Name": "Transforms.TextFeaturizer",
						"Inputs": {
							"Column": {
								"Name": "FeaturesText",
								"Source": [
									"SentimentText"
								]
							},
							"Data": "$Copy_DataView_Output_1528237424459_579",
							"WordFeatureExtractor": {
								"Name": "NGram",
								"Settings": {
									"NgramLength": 2,
									"SkipLength": 0,
									"AllLengths": true,
									"MaxNumTerms": [
										10000000
									],
									"Weighting": "Tf"
								}
							},
							"CharFeatureExtractor": {
								"Name": "NGram",
								"Settings": {
									"NgramLength": 3,
									"SkipLength": 0,
									"AllLengths": true,
									"MaxNumTerms": [
										10000000
									],
									"Weighting": "Tf"
								}
							}
						},
						"Outputs": {
							"OutputData": "$Text_DataView_Output_1528237424462_956",
							"Model": "$Text_TransformModel_Output_1528237424462_164"
						}
					},
					{
						"Name": "Transforms.ColumnConcatenator",
						"Inputs": {
							"Column": [
								{
									"Name": "Features",
									"Source": [
										"FeaturesText"
									]
								}
							],
							"Data": "$Text_DataView_Output_1528237424462_956"
						},
						"Outputs": {
							"OutputData": "$Concat_DataView_Output_1528237424464_228",
							"Model": "$Concat_TransformModel_Output_1528237424464_5"
						}
					},
					{
						"Name": "Trainers.AveragedPerceptronBinaryClassifier",
						"Inputs": {
							"TrainingData": "$Concat_DataView_Output_1528237424464_228",
							"LossFunction": {
								"Name": "HingeLoss",
								"Settings": {
									"Margin": 1
								}
							},
							"NumIterations": 10,
							"Calibrator": {
								"Name": "PlattCalibrator"
							}
						},
						"Outputs": {
							"PredictorModel": "$ap_PredictorModel_Output_1528237424465_314"
						}
					},
					{
						"Name": "Transforms.ManyHeterogeneousModelCombiner",
						"Inputs": {
							"TransformModels": [
								"$Copy_TransformModel_Output_1528237424459_493",
								"$Text_TransformModel_Output_1528237424462_164",
								"$Concat_TransformModel_Output_1528237424464_5"
							],
							"PredictorModel": "$ap_PredictorModel_Output_1528237424465_314"
						},
						"Outputs": {
							"PredictorModel": "$PredictorModel0"
						}
					}
				],
				"Inputs": {
					"Data": "$data0"
				},
				"Outputs": {
					"PredictorModel": "$PredictorModel0"
				}
			},
			"Outputs": {
				"PredictorModel": "$PredictorModel0",
				"Warnings": "$Warnings0",
				"OverallMetrics": "$OverallMetrics0",
				"PerInstanceMetrics": "$PerInstanceMetrics0",
				"ConfusionMatrix": "$ConfusionMatrix0"
			}
		}
	]
}
```

**Comment by jwood803** [comment]
@sfilipi I took a look at this and can reproduce with a test. Debugging through, it looks like here is where it is failing:

https://github.com/dotnet/machinelearning/blob/c023727b76970ab913ec1ce38276508835c17bcf/src/Microsoft.ML/Runtime/EntryPoints/JsonUtils/ExecuteGraphCommand.cs#L149

In there it has a contract to check if `idv` is null and throws if it isn't. I've added this to it:

```
if (idv != null)
{
    SaveDataView(idv, path, extension);
}
else
{
    // Add warning            
}
```

However, I'm still going through to see how I can output a warning message.
